Feature/category tabs and products grid

MohamedOmar989 opened the pull request:
<img width="565" height="1180" alt="88" src="https://github.com/user-attachments/assets/61bfd4f0-3b87-467e-98c1-cdac5a8a5ae6" />

MohamedOmar989 committed:
Merge remote-tracking branch 'origin/develop' into feature/Category-Tabs-and-Products-Grid

MohamedOmar989 committed:
Merge remote-tracking branch 'origin/develop' into feature/Category-Tabs-and-Products-Grid

MohamedOmar989 committed:
Merge remote-tracking branch 'origin/feature/home-screen-design' into feature/Category-Tabs-and-Products-Grid

MohamedOmar989 committed:
feat: implement category tabs and update chip

MohamedOmar989 committed:
feat: Implement favorite functionality and products list

MohamedOmar989 committed:
refactor: extract widgets from HomeScreen

MohamedOmar989 committed:
feat: add category icons

MohamedOmar989 committed:
feat: add new category and favorite icons

M-Abdelnabi reviewed (approved):
Looks good for me 
good job omar 👏👏

baraa0abd commented on lib/core/design_system/widgets/ChipCategory.dart:
```diff
@@ -35,15 +39,25 @@ class ChipCategory extends StatelessWidget {
+                    child: SvgPicture.asset(
+                      assetIcon!,
+                      width: 18,
+                      height: 18,
```
I think in figma the it's 20 

<img width="84" height="102" alt="Image" src="https://github.com/user-attachments/assets/02c2405b-9e5d-443c-921b-0307de40f90a" />

baraa0abd reviewed (changes requested):
the background color of the cards are different than figma design ( the background color covers the whole screen instead of the product card only )  : 

## in the pic ##

<img width="178" height="305" alt="Image" src="https://github.com/user-attachments/assets/8970993c-9bdd-43f8-846d-85e2f4662701" />

## in figma design ##

<img width="579" height="536" alt="Image" src="https://github.com/user-attachments/assets/0493c763-53fd-4d3d-a159-32e96c5de5bd" />

mohamedebrahem13 reviewed (changes requested):
screen colors is deferent in Figma and this make some component appears without background

abdo-essam committed:
Merge branch 'refs/heads/feature/home-screen-design' into feature/Category-Tabs-and-Products-Grid

# Conflicts:
#	lib/core/design_system/constants/assets.dart
#	lib/ui/screens/home/home_screen.dart

abdo-essam committed:
refactor(ui): Refactor home screen and widgets

Move chip category and refactor home screen widgets for better organization and maintainability.

abdo-essam committed:
feat(ui): Implement category tabs and product grid

Implements the category tabs and product grid on the
home screen. Includes product cards and images.
